File: FastMines_android/src/main/java/fmg/android/app/model/dataSource/MosaicSkillDataSource.java
package fmg.android.app.model.dataSource;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import fmg.android.app.ProjSettings;
import fmg.android.app.model.items.MosaicSkillDataItem;
import fmg.android.img.AndroidBitmapView;
import fmg.android.img.MosaicSkillImg2;
import fmg.common.Color;
import fmg.common.geom.BoundDouble;
import fmg.core.img.ImageHelper;
import fmg.core.img.MosaicSkillModel2;
import fmg.core.types.ESkillLevel;

public class MosaicSkillDataSource extends BaseDataSource<
        MosaicSkillDataItem, ESkillLevel, MosaicSkillModel2, AndroidBitmapView<MosaicSkillModel2>, MosaicSkillImg2.MosaicSkillAndroidBitmapController,
        MosaicSkillDataItem, ESkillLevel, MosaicSkillModel2, AndroidBitmapView<MosaicSkillModel2>, MosaicSkillImg2.MosaicSkillAndroidBitmapController>
{

    @Override
    public MosaicSkillDataItem getHeader() {
        if (header == null) {
            header = new MosaicSkillDataItem(null);

            var ctrlr = header.getEntity();
            var model = ctrlr.getModel();
            model.setPadding(new BoundDouble(3));
            model.setBackgroundColor(Color.Transparent());
            ctrlr.setFps(20);
            ctrlr.setAnimatePeriod(13000);
            ctrlr.setPolarLightsForeground(true);

            firePropertyChanged(PROPERTY_HEADER);
        }
        return header;
    }

    @Override
    public List<MosaicSkillDataItem> getDataSource() {
        if (!isDisposed() && ((dataSource == null) || dataSource.isEmpty())) {
            dataSource = Stream.of(ESkillLevel.values())
                    .filter(sk ->
                        !ProjSettings.isReleaseMode() || (sk != ESkillLevel.eCustom)
                    )
                    .map(MosaicSkillDataItem::new)
                    .peek(item -> {
                        var ctrlr = item.getEntity();
                        ctrlr.setFps(20);
                        ctrlr.setAnimatePeriod(3600);
                        applySelection(item);
                    })
                    .collect(Collectors.toList());

            firePropertyChanged(PROPERTY_DATA_SOURCE);
        }
        return dataSource;
    }

    @Override
    protected void onCurrentItemChanged() {
        getDataSource().forEach(this::applySelection);
    }

    /** for one selected item - start animate; for all other - stop animate */
    private void applySelection(MosaicSkillDataItem item) {
        boolean selected = (item.getUniqueId().ordinal() == currentItemPos);
        var ctrlr = item.getEntity();
        var model = ctrlr.getModel();
        ctrlr.setPolarLightsForeground(selected);
        model.setBorderColor(selected ? Color.LawnGreen() : Color.IndianRed());
        model.setBackgroundColor(selected ? ImageHelper.DEFAULT_BK_COLOR : DefaultBkColor);
        model.setPadding(new BoundDouble(selected ? 5 : 15));
        if (!selected)
            model.setForegroundColor(ImageHelper.DEFAULT_FOREGROUND_COLOR.brighter());
//        else {
//            HSV hsv = new HSV(AnimatedImageModel.DefaultForegroundColor);
//            hsv.s = hsv.v = 100;
//            model.setForegroundColor(hsv.toColor());
//        }
    }

}

File: FastMines_uwp/src/fmg/uwp/app/Model/DataSources/MosaicSkillDataSource.cs
using System.Linq;
using System.Collections.ObjectModel;
using Fmg.Common;
using Fmg.Common.Geom;
using Fmg.Core.Img;
using Fmg.Core.Types;
using Fmg.Core.Mosaic;
using Fmg.Uwp.App.Model.Items;
using MosaicSkillView       = Fmg.Uwp.Img.Win2d.MosaicSkillImg.CanvasBmpView;
using MosaicSkillController = Fmg.Uwp.Img.Win2d.MosaicSkillImg.CanvasBmpController;

namespace Fmg.Uwp.App.Model.DataSources {

    /// <summary> DataSource menu items (mosaic skills) </summary>
    public class MosaicSkillDataSource : BaseDataSource<
        MosaicSkillDataItem, ESkillLevel?, MosaicSkillModel, MosaicSkillView, MosaicSkillController,
        MosaicSkillDataItem, ESkillLevel?, MosaicSkillModel, MosaicSkillView, MosaicSkillController>
    {

        public override MosaicSkillDataItem Header {
            get {
                if (header == null) {
                    header = new MosaicSkillDataItem(null);

                    var model = header.Entity.Model;
                    model.Padding = new BoundDouble(3);
                    model.BackgroundColor = Color.Transparent;
                    model.TotalFrames = 260;
                    model.AnimatePeriod = 12900;
                    model.PolarLights = true;
                    model.Animated = true;
                }
                return header;
            }
        }

        public override ObservableCollection<MosaicSkillDataItem> DataSource {
            get {
                if (!dataSource.Any() && !Disposed) {
                    foreach (var e in ESkillLevelEx.GetValues()) {
                        var item = new MosaicSkillDataItem(e);
                        var model = item.Entity.Model;
                        model.TotalFrames = 72;
                        model.AnimatePeriod = 3600;
                        ApplySelection(item);
                        dataSource.Add(item);
                    }
                    _notifier.FirePropertyChanged();
                }
                return dataSource;
            }
        }

        protected override void OnCurrentItemChanged() {
            foreach (var item in DataSource)
                ApplySelection(item);
        }

        /// <summary> for one selected item - start animate; for all other - stop animate </summary>
        private void ApplySelection(MosaicSkillDataItem item) {
            var selected = ReferenceEquals(item, CurrentItem);
            var model = item.Entity.Model;
            model.Animated = selected;
            model.PolarLights = selected;
            model.BorderColor = selected ? Color.LawnGreen : Color.IndianRed;
            model.BackgroundColor = selected ? AnimatedImageModelConst.DefaultBkColor : DefaultBkColor;
            model.Padding = new BoundDouble(selected ? 5 : 15);
            if (!selected)
                model.ForegroundColor = AnimatedImageModelConst.DefaultForegroundColor.Brighter();
            else
                model.ForegroundColor = Color.Orchid;
            //else {
            //    HSV hsv = new HSV(ImageModelConsts.DefaultForegroundColor);
            //    hsv.s = hsv.v = 100;
            //    model.ForegroundColor = hsv.ToColor();
            //}
        }

    }

}
